docs/reference/superjson.mdx

```diff
@@ -0,0 +1,294 @@
+
+- `kind`: `none` or `circuit-breaker`
+- `maxContiguousRetries`: maximum number of retry attempts, positive integer, default 5
+- `requestTimeout`: timeout after which a request fails, positive integer, milliseconds, default 30000
+- `backoff`: either string `exponential` or an object with following properties:
+    - `kind`: string `exponential`
+    - `start`: the initial delay time, optional, positive integer, milliseconds, default 2000
+    - `factor`: the factor by which is the delay time multiplied, optional, positive integer, default 2
+
+```json
+{
+  "kind": "circuit-breaker",
+  "maxContiguousRetries": 3,
+  "requestTimeout": 60000,
+  "backoff": {
+    "kind": "exponential",
+    "start": 1000,
+    "factor": 3
+  }
+}
+```
+
+## Providers
+
+The providers section consists of a record of all used providers. The key is the name of the provider and the value is either a file URI or a [ProviderSettings](#providersettings) object.
+
+```json
+{
+  "providers": {
+    "provider1": "file:path/to/provider.json",
+    "provider2": {
+      // ...
+    }
+  }
+}
+```
+
+### ProviderSettings
+
+The ProviderSettings object has following optional properties:
+
+- `file`: a file URI string
+- `security`: an array of [SecurityValues](#securityvalues) objects
+
+```json
+{
+  "file": "file:path/to/provider.json",
+  "security": [
+      {
+        // ...
+      }
+  ]
+}
+```
+
+### SecurityValues
+
+SecurityValues is an union of several types of security configuration objects: API Key, Basic Auth, Bearer Token or Digest. All of them have a property in common:
+
+- `id`: identifier of the security scheme, referenced in map
+
+#### API Key 
+API key uses provided API key in a header, body, path or query string (specified in provider.json)
+
+- `apikey`: string containing API key
+
+#### Basic Auth
+Basic Auth uses provided username and password and sends them base64 encoded in Authorization header as per [RFC 7617](https://datatracker.ietf.org/doc/html/rfc7617)
+
+- `username`: string containing user name
+- `password`: string containing password
+
+#### Bearer Token
+Bearer Token uses provided token and sends it in Authorization header as per [RFC 6750](https://datatracker.ietf.org/doc/html/rfc6750)
+
+- `token`: string containing bearer token
+
+#### Digest
+Digest uses provided token and sends it in Authorization header as per [RFC 2617](https://datatracker.ietf.org/doc/html/rfc2617).
+*This is not implemented yet.*
+
+- `digest`: string containing digest
+
+```json
+[
+  {
+    "id": "first",
+    "apikey": "my-api-key"
+  },
+  {
+    "id": "second",
+    "username": "myname",
+    "password": "mypass"
+  },
+  {
+    "id": "third",
+    "token": "my-bearer-token"
+  },
+  {
+    "id": "fourth",
+    "digest": "my-digest"
+  }
+]
+```
+
+## Example
```

Comment:
Consider moving the example at the start of the document, optionally with comments annotating the specific parts. This would be probably more helpful than the current "Anatomy" example.